src/osd/PrimaryLogPG.h

```diff
@@ -692,7 +692,17 @@ class PrimaryLogPG : public PG, public PGBackend::Listener {
   public:
     hobject_t hoid;
     OpRequestRef op;
-    xlist<RepGather*>::item queue_item;
+  private:
+    boost::intrusive::list_member_hook<> queue_item_;
+  public:
+    typedef boost::intrusive::list<
+      RepGather,
+      boost::intrusive::member_hook<
+        RepGather,
+        boost::intrusive::list_member_hook<>,
+        &RepGather::queue_item_> > queue_item_t;
+    queue_item_t queue_item;
```

Comment:
what you call `queue_item_t` here is the list type itself - can we call it `queue_t` instead?

with that in mind, it doesn't make sense for a queue item (`RepGather`) to itself contain a queue, so this unused member variable `queue_item_t queue_item;` should be removed